src/app/public/modules/inline-delete/inline-delete-adapter.service.ts

```diff
@@ -0,0 +1,134 @@
+import {
+  Injectable,
+  Renderer2,
+  RendererFactory2
+} from '@angular/core';
+
+/**
+ * @internal
+ */
+@Injectable()
+export class SkyInlineDeleteAdapterService {
+
+  private element: HTMLElement;
+  private focussableElements: HTMLElement[];
+  private parentEl: HTMLElement;
+  private parentElUnlistenFn: Function;
+  private renderer: Renderer2;
+
+  constructor(
+    private rendererFactory: RendererFactory2
+  ) {
+    this.renderer = this.rendererFactory.createRenderer(undefined, undefined);
```

Comment:
Any reason you don't just inject the `Renderer2` in the constructor? Why use a factory?

Reply:
You can't inject `Renderer2` directly into services. When using it in a service you have to use the factory. We do this in all of our adapter services.